orangecontrib/spectroscopy/widgets/preprocessors/spikeremoval.py

```diff
@@ -0,0 +1,45 @@
+from PyQt5.QtWidgets import QVBoxLayout
+from Orange.widgets import gui
+from orangecontrib.spectroscopy.widgets.preprocessors.utils import \
+    BaseEditorOrange
+from orangecontrib.spectroscopy.preprocess import Despike
+
+class SpikeRemovalEditor(BaseEditorOrange):
+    """
+    Spike Removal.
+    """
+
+    def __init__(self, parent=None, **kwargs):
+        super().__init__(parent, **kwargs)
+        self.dis = 5
+        self.threshold = 7
+        self.cutoff = 100
+        self.controlArea.setLayout(QVBoxLayout())
+        gui.spin(self.controlArea, self, "cutoff", label="Peak difference cutoff", minv=0,
```

Comment:
This cutoff is too limiting. For normalized data, the lowest value of 1 is not enough. It should allow floating point numbers.

Reply:
I'll change it to allow for hundredths values to be input, I didn't think about the need for as specific cutoffs but it would be useful to have.

Reply:
Hundreths again present some arbitrary limit. Please do not limit it at all, except for the limit that it is non-negative. I think 0 would also present a valid cutoff for the algorithm to work and might be useful with some (high) z-score thresholds.

I'd also avoid using spinboxes. For non-integer inputs free form text windows are usually a better fit. Check, for example, `lineEditDecimalOrNone` from `orangecontrib.spectroscopy.widgets.gui` (it can have limits and default values).

The same might actually go for the z-score limit, but there it is probably not as important. There hundreths (tens?) might be enough.

Reply:
Oh, ignore my last change for the cutoff, I'll use the lineEditDecimalOrNone since it seems more robust. Z-score I agree would be fine with hundreths or tens but I'll also do the same for it as the cutoff to not limit it.

Reply:
I've changed them to the `lineEditDecimalOrNone ` windows, I've not used them before, but I believe I put them in correctly.